plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -59,28 +61,50 @@
         }
       });
 
+  private static AnalyticsPingManager instance;
+
+  // Preference store (should be ConfigurationScope-d) from which we get UUID, opt-in status, etc.
+  private IEclipsePreferences preferences;
+
+  private OptInDialogCreator optInDialogCreator;
+
+  @VisibleForTesting
+  AnalyticsPingManager(
+      IEclipsePreferences preferences, OptInDialogCreator optInDialogCreator) {
+    this.preferences = preferences;
+    this.optInDialogCreator = optInDialogCreator;
+  }
+
+  public static AnalyticsPingManager getInstance() {
```

Comment:
That's a good question. I thought it is tolerable to create two and overwrite the previous one without synchronization. However, if we are really concerned about absolute correctness in concurrency, I think not only this instance but also every access to the preference store (which is sort of a shared repository) might need to be synchronized. But then, I think concurrent access to the preference store is also tolerable. Your thoughts? I will follow what we finally have consensus on.